moped-editor/src/views/staff/StaffListViewQueryConf.js

```diff
@@ -16,14 +21,37 @@ export const StaffListViewQueryConf = {
     },
   },
   table: "moped_users",
-  single_item: "/moped/staff/edit",
-  new_item: "/moped/staff/new",
-  new_item_label: "Add Staff",
+  customSingleItemButton: id => (
+    <Can
+      perform="user:edit"
+      yes={
+        <RouterLink to={`/moped/staff/edit/${id}`}>
+          <EditIcon color="primary" />
+        </RouterLink>
+      }
+    />
+  ),
```

Comment:
This is again very brilliant! Frankly, I dislike to see components in a configuration.

This should stay here for now, but we need to figure out a way to control this pattern.

Maybe we could have a section for custom components, maybe something like this:

```

import { StaffListViewCustomComponents } from "./StaffListViewCustomComponents";

{
...
customComponents: StaffListViewCustomComponents,
...
}

```

Custom Components file:

```
export const StaffListViewCustomComponents =  {
singleItemButton: id => (
  <Can
      perform="user:edit"
      yes={
        <RouterLink to={`/moped/staff/edit/${id}`}>
          <EditIcon color="primary" />
        </RouterLink>
      }
    />
  ),
  newItemButton: ewItemButton: (
    <Can
      perform="user:create"
      yes={
        <Button
          color="primary"
          variant="contained"
          component={RouterLink}
          to={"/moped/staff/new"}
          startIcon={<Icon>add_circle</Icon>}
        >
          Add Staff
        </Button>
      }
    />
  )
}
```

Reply:
I like this a lot, Sergio. We might even need to recycle this component somewhere down the line. I'm going to update and pull that code into a separate file before merging. 🙏